app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,509 @@
+            let shadow     = shadow::from_shape_with_alpha(base_shape.into(),&alpha,style);
+
+            (shadow + background + divider).into()
+        }
+    }
+}
+
+mod hline {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        (style:Style) {
+            let width            = Var::<Pixels>::from("input_size.x");
+            let height           = Var::<Pixels>::from("input_size.y");
+
+            let rect = Rect((width,height));
+            let rect = rect.fill(color::Rgb::from_css_hex(SECTION_HEADING_COLOR_HEX).unwrap());
+            rect.into()
+        }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+
+/// The Model of Select Component.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Model {
+    app:                 Application,
+    logger:              Logger,
+    display_object:      display::object::Instance,
+    background:          background::View,
+    scroll_area:         ScrollArea,
+    favourites_section:  ColumnSection,
+    local_scope_section: WideSection,
+    sub_modules_section: ColumnSection,
+    layers:              Layers,
+}
+
+impl Model {
+    fn new(app: &Application) -> Self {
+        let logger = Logger::new("ComponentBrowserPanel");
+        let app = app.clone_ref();
+        let display_object = display::object::Instance::new(&logger);
+
+        let background = background::View::new(&logger);
+        background.bg_color.set(BACKGROUND_COLOR.into());
+        background.size.set(Vector2::new(WIDTH, HEIGHT));
+        display_object.add_child(&background);
+        app.display.default_scene.layers.below_main.add_exclusive(&background);
+
+        let favourites_section = Self::init_column_section(&app, FAVOURITES_SECTION_HEADING_LABEL);
+        let local_scope_section = Self::init_wide_section(&app, LOCAL_SCOPE_SECTION_HEADING_LABEL);
+        local_scope_section.content.set_color(FAVOURITES_SECTION_BASE_COLOR);
+        let sub_modules_section =
+            Self::init_column_section(&app, SUB_MODULES_SECTION_HEADING_LABEL);
+
+        let scroll_area = ScrollArea::new(&app);
+        display_object.add_child(&scroll_area);
+        scroll_area.resize(Vector2::new(
+            CONTENT_WIDTH - PADDING_INNER,
+            CONTENT_HEIGHT - 2.0 * PADDING_INNER,
+        ));
+        scroll_area.set_position_xy(Vector2::new(
+            -CONTENT_WIDTH / 2.0,
+            CONTENT_HEIGHT / 2.0 - MENU_HEIGHT / 2.0,
+        ));
+        scroll_area.set_corner_radius_bottom_right(CORNER_RADIUS);
+
+        let camera = &scroll_area.content_layer().camera();
+        let parent_layer = scroll_area.content_layer();
+        let layers = component_group::Layers::new(&app.logger, camera, parent_layer);
+
+        favourites_section.set_parent(scroll_area.content());
+        local_scope_section.set_parent(scroll_area.content());
+        sub_modules_section.set_parent(scroll_area.content());
+
+        // Required for correct clipping.
```

Comment:
This comment does not explain anything. When reading the code, I don't understand it. Why is it required? What happens with the clipping if its not there? What does it mean "correct" in this context? How the incorrect clipping looks like without these lines?